packages/medusa/src/services/cart.ts

```diff
@@ -2163,10 +2188,31 @@ class CartService extends TransactionBaseService {
           const lineItemServiceTx =
             this.lineItemService_.withTransaction(transactionManager)
 
+          let productShippinProfileMap = new Map<string, string>(
+            cart.items.map((item) => [
+              item.variant?.product?.id,
+              item.variant?.product?.profile_id,
+            ])
+          )
+
+          if (
+            this.featureFlagRouter_.isFeatureEnabled(
+              IsolateProductDomainFeatureFlag.key
+            )
+          ) {
+            productShippinProfileMap =
+              await this.shippingProfileService_.getMapProfileIdsByProductIds(
+                cart.items.map((item) => item.variant.product_id)
+              )
+          }
```

Comment:
suggestion: Can we save a mapping here?
```suggestion
          let productShippinProfileMap = new Map<string, string>()

          if (
            this.featureFlagRouter_.isFeatureEnabled(
              IsolateProductDomainFeatureFlag.key
            )
          ) {
            productShippinProfileMap =
              await this.shippingProfileService_.getMapProfileIdsByProductIds(
                cart.items.map((item) => item.variant.product_id)
              )
          } else {
            productShippinProfileMap = cart.items.map((item) => [
              item.variant?.product?.id,
              item.variant?.product?.profile_id,
            ])
          }
```